crates/noirc_evaluator/src/ssa_refactor/ir/instruction.rs

```diff
@@ -329,7 +339,32 @@ impl Instruction {
                     None
                 }
             }
-            Instruction::Call { .. } => None,
+            Instruction::Call { func, arguments } => {
+                let intrinsic = match &dfg[*func] {
+                    Value::Intrinsic(intrinsic) => *intrinsic,
+                    _ => return None,
+                };
+                let constant_args: Option<Vec<_>> =
+                    arguments.iter().map(|value_id| dfg.get_numeric_constant(*value_id)).collect();
```

Comment:
Since Cast and Call optimizations are getting more involved, can we break them out into separate functions?

Reply:
Done